geomesa-spark/geomesa-spark-geotools/src/main/scala/org/locationtech/geomesa/spark/geotools/GeoToolsSpatialRDDProvider.scala

```diff
@@ -0,0 +1,108 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.spark.geotools
+
+import java.io.{IOException, ObjectInputStream, ObjectOutputStream, Serializable}
+import java.util
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.rdd.RDD
+import org.geotools.data.{DataStoreFinder, Query, Transaction}
+import org.locationtech.geomesa.compute.spark.GeoMesaSparkKryoRegistrator
+import org.locationtech.geomesa.jobs.GeoMesaConfigurator
+import org.locationtech.geomesa.spark.SpatialRDDProvider
+import org.locationtech.geomesa.utils.geotools.Conversions._
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.JavaConversions._
+import scala.util.control.NonFatal
+
+class GeoToolsSpatialRDDProvider extends SpatialRDDProvider with LazyLogging {
+  override def canProcess(params: util.Map[String, Serializable]): Boolean = {
+    params.containsKey("geotools") && DataStoreFinder.getDataStore(params) != null
```

Comment:
Nope, but we could re-work the code a little bit to call canProcess directly instead.